core/store/src/trie/state_snapshot.rs

```diff
@@ -367,19 +348,11 @@ impl ShardTries {
         let _span =
             tracing::info_span!(target: "state_snapshot", "maybe_open_state_snapshot").entered();
         metrics::HAS_STATE_SNAPSHOT.set(0);
-        let StateSnapshotConfig {
-            state_snapshot_type,
-            home_dir,
-            hot_store_path,
-            state_snapshot_subdir,
-            compaction_enabled: _,
-        } = self.state_snapshot_config();
-        if state_snapshot_type == &StateSnapshotType::ForReshardingOnly {
```

Comment:
@nikurt, wanted to confirm, it should be fine to remove this check from maybe_open_state_snapshot right? This function is called while creating ShardTries and logs an error in case snapshot doesn't exist.

Reply:
Yes, should be fine to always attempt opening a state snapshot.